Drop error log for "job scaling blocked due to active deployment"

Just a slight quality-of-life request - can logs for this be dropped to WARN or maybe even INFO? 

To me it seems this is expected behavior, and wouldn't be considered an error. It would be nice to drop the severity on this one so alerts for real errors won't trip over this and it won't have to be filtered otherwise.

```
2021-07-19T16:37:34.048Z [ERROR] policy_eval.worker: failed to evaluate policy: eval_id=70c8ea00-1c40-9ffb-f732-29cddb212153 eval_token=8e13fa3c-8c68-2aa8-da50-be561332f67e id=3fe0dfca-376c-6a37-deba-dd84d95e9a2b policy_id=0b03eba3-153e-4c7f-dab9-325d3cff760e queue=horizontal error="failed to scale target: failed to scale group /: Unexpected response code: 400 (job scaling blocked due to active deployment)"
```

Low priority - just a thought. Thanks folks!

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,7 @@
 
 IMPROVEMENTS:
  * agent: Dispense `fixed-value`, `pass-through`, and `threshold` strategy plugins by default [[GH-536](https://github.com/hashicorp/nomad-autoscaler/pull/536)]
+ * plugins/target/nomad: Reduce log level for active deployments error messages [[GH-542](https://github.com/hashicorp/nomad-autoscaler/pull/542)]
  * policy: Prevent scaling cluster to zero when using the Nomad APM [[GH-534](https://github.com/hashicorp/nomad-autoscaler/pull/534)]
  * scaleutils: Add combined filter to allow filtering by node class and datacenter [[GH-535](https://github.com/hashicorp/nomad-autoscaler/pull/535)]
 

--- a/plugins/builtin/target/nomad/plugin/plugin.go
+++ b/plugins/builtin/target/nomad/plugin/plugin.go
@@ -2,6 +2,7 @@ package nomad
 
 import (
 	"fmt"
+	"strings"
 	"sync"
 	"time"
 
@@ -135,6 +136,12 @@ func (t *TargetPlugin) Scale(action sdk.ScalingAction, config map[string]string)
 		&q)
 
 	if err != nil {
+		// Active deployments errors are fairly common and usually not
+		// impactful to the target's eventual end state, so special case them
+		// to return a no-op error instead.
+		if strings.Contains(err.Error(), "job scaling blocked due to active deployment") {
+			return sdk.NewTargetScalingNoOpError("skipping scaling group %s/%s due to active deployment", config[configKeyJobID], config[configKeyGroup])
+		}
 		return fmt.Errorf("failed to scale group %s/%s: %v", config[configKeyJobID], config[configKeyGroup], err)
 	}
 	return nil

--- a/policyeval/base_worker.go
+++ b/policyeval/base_worker.go
@@ -214,6 +214,11 @@ func (w *BaseWorker) handlePolicy(ctx context.Context, eval *sdk.ScalingEvaluati
 	// handler understand what do to.
 	err = w.runTargetScale(targetInst, eval.Policy, *winningAction)
 	if err != nil {
+		if _, ok := err.(*sdk.TargetScalingNoOpError); ok {
+			logger.Info("scaling action skipped", "reason", err)
+			return nil
+		}
+
 		metrics.IncrCounter([]string{"scale", "invoke", "error_count"}, 1)
 		return fmt.Errorf("failed to scale target: %v", err)
 	} else {

--- a/sdk/target.go
+++ b/sdk/target.go
@@ -1,5 +1,29 @@
 package sdk
 
+import (
+	"fmt"
+)
+
+// TargetScalingNoOp is a special error type that can be used by target plugins
+// to indicate that a scaling request didn't result in any action, but didn't
+// fail either.
+// This can be used to avoid post-scaling actions such as placing the policy in
+// cooldown.
+type TargetScalingNoOpError struct {
+	Err error
+}
+
+// NewTargetScalingNoOpError returns a new target scaling no-op error with the
+// provided formatted message.
+func NewTargetScalingNoOpError(msg string, args ...interface{}) *TargetScalingNoOpError {
+	return &TargetScalingNoOpError{Err: fmt.Errorf(msg, args...)}
+}
+
+// Error implements the error interface.
+func (n *TargetScalingNoOpError) Error() string {
+	return n.Err.Error()
+}
+
 // TargetStatus is the response object when performing the Status call of the
 // target plugin interface. The response details key information about the
 // current state of the target.

--- a/sdk/target.go
+++ b/sdk/target.go
@@ -4,9 +4,9 @@ import (
 	"fmt"
 )
 
-// TargetScalingNoOp is a special error type that can be used by target plugins
-// to indicate that a scaling request didn't result in any action, but didn't
-// fail either.
+// TargetScalingNoOpError is a special error type that can be used by target
+// plugins to indicate that a scaling request didn't result in any action, but
+// didn't fail either.
 // This can be used to avoid post-scaling actions such as placing the policy in
 // cooldown.
 type TargetScalingNoOpError struct {

